src/core/opamStd.ml

```diff
@@ -1004,8 +1004,9 @@ module OpamSys = struct
                   f a
               else
                 f a
-          | exception _ ->
+          | exception e ->
               Unix.close_process_full process |> ignore;
+              fatal e;
```

Comment:
This should be first, I think - I think `Unix.close_process_full` can also raise?